vsintegration/src/FSharp.Editor/Navigation/NavigateToSignatureMetadataService.fs

```diff
@@ -0,0 +1,311 @@
+﻿namespace Microsoft.VisualStudio.FSharp.Editor
+
+open System
+open System.IO
+open System.Text
+open System.Security
+open System.Composition
+open System.Collections.Generic
+open Microsoft.VisualStudio
+open Microsoft.VisualStudio.Shell.Interop
+open Microsoft.VisualStudio.Shell
+open Microsoft.FSharp.Compiler.SourceCodeServices
+open Microsoft.FSharp.Compiler.Range
+open Microsoft.CodeAnalysis
+open Microsoft.CodeAnalysis.Text
+open System.ComponentModel.Composition
+open Microsoft.VisualStudio.FSharp.Editor
+open Microsoft.CodeAnalysis.Formatting
+open Microsoft.VisualStudio.ComponentModelHost
+open Microsoft.VisualStudio.Editor
+open Microsoft.VisualStudio.TextManager.Interop
+open Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem 
+
+[<Export>][<Shared>]
+type internal NavigateToSignatureMetadataService [<ImportingConstructor>] 
+    (   checkerProvider: FSharpCheckerProvider
+    ,   projectInfoManager: ProjectInfoManager
+    ,   workspace : VisualStudioWorkspaceImpl
+    ) =
+    
+    let xmlDocCache = Dictionary<string, IVsXMLMemberIndex>()
+    let serviceProvider =  ServiceProvider.GlobalProvider                
+    let xmlIndexService = serviceProvider.GetService<SVsXMLMemberIndexService,IVsXMLMemberIndexService>()
+    let componentModel = serviceProvider.GetService<SComponentModel,IComponentModel>()
+    let editorAdapterFactory = componentModel.GetService<IVsEditorAdaptersFactoryService>()
+
+    
+    /// Stores the editor window displaying signature metadata if one has already been opened.
+    /// Ensures that only one editor window is open for signature metadata at a time.
+    let mutable currentWindow: IVsWindowFrame option = None 
+
+    // Close the signature metadata window with the solution
+    do workspace.WorkspaceChanged.Add (fun workspaceChange ->
+       match workspaceChange.Kind with 
+       | WorkspaceChangeKind.SolutionCleared
+       | WorkspaceChangeKind.SolutionReloaded
+       | WorkspaceChangeKind.SolutionRemoved ->
+            currentWindow
+            |> Option.bind Option.ofNull
+            |> Option.iter (fun window -> window.CloseFrame (uint32 __FRAMECLOSE.FRAMECLOSE_NoSave) |> ignore)
+       | _ -> ()
+       )
+
+    /// If the XML comment starts with '<' not counting whitespace then treat it as a literal XML comment.
+    /// Otherwise, escape it and surround it with <summary></summary>
+    let processXml (xml: string) =
+        if String.IsNullOrEmpty xml then xml else
+        let trimmedXml = xml.TrimStart([|' ';'\r';'\n'|])
+        if String.IsNullOrEmpty trimmedXml then xml else
+        match trimmedXml.[0] with
+        | '<' ->    String.Join ("", "<root>", xml, "</root>")
+        | _   ->    let escapedXml = SecurityElement.Escape xml
+                    String.Join ("", "<summary>", escapedXml, "</summary>")
```

Comment:
@vasily-kirichenko I've seen this style of indentation when first branch is simple, and the code will need further cases added, I think it makes sense.

See https://github.com/smoothdeveloper/visualfsharp/blob/a3a3fb552d27ad8ba5cab6497a033ebf19655025/src/fsharp/NicePrint.fs#L1965-L2011 and try to indent the stuff, as you'll add cases, you'll need to go deeper on indentation and it will look really odd.